### PR TITLE
Amplify Mock: Add note about JDK issues on Apple Silicon

### PR DESCRIPTION
**Pull request by danabrams**
_Issue #, if available:_ [7411 on Amplify-CLI](https://github.com/aws-amplify/amplify-cli/issues/7411)

_Description of changes:_
Added a note describing the fix for this issue on Apple Silicon (M1) Macs

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/cli/usage/mock.md [hunk 1]
```diff
@@ -15,6 +15,10 @@ In order to quickly test and debug without pushing all changes in your project t
 Java is required on your development workstation to use Local Mocking in Amplify
 </amplify-callout>
 
+> Currently, on Apple Silicon Macs, Amplify sometimes fails to start mocks when using certain JDK versions built for Arm processors.
```
Comment (josefaidt) [inline docs/cli/usage/mock.md:18]:
```suggestion
> **NOTE:** Currently, on Apple Silicon Macs, Amplify sometimes fails to start mocks when using certain JDK versions built for ARM processors. 
```

File: docs/cli/usage/mock.md [hunk 2]
```diff
@@ -15,6 +15,10 @@ In order to quickly test and debug without pushing all changes in your project t
 Java is required on your development workstation to use Local Mocking in Amplify
 </amplify-callout>
 
+> Currently, on Apple Silicon Macs, Amplify sometimes fails to start mocks when using certain JDK versions built for Arm processors. 
+>
+> If you encounter [this issue](https://github.com/aws-amplify/amplify-cli/issues/7411), please try using the official openJDK 16.0.1 from the Java website: [JDK Download](https://github.com/aws-amplify/amplify-cli/issues/7411#:~:text=https%3A//download.java.net/java/GA/jdk16.0.1/7147401fd7354114ac51ef3e1328291f/9/GPL/openjdk-16.0.1_osx-x64_bin.tar.gz)
```
Comment (josefaidt) [inline docs/cli/usage/mock.md:20]:
```suggestion
> If you encounter [this issue](https://github.com/aws-amplify/amplify-cli/issues/7411), please try using the official openJDK 16.0.1 from the Java website: [JDK Download](https://jdk.java.net/16/)
```

File: docs/cli/usage/mock.md (hunk 1 above)
Comment (josefaidt) [reply docs/cli/usage/mock.md:18]: Adding `NOTE` will adhere to other instances of blockquotes with notes

File: docs/cli/usage/mock.md (hunk 2 above)
Comment (josefaidt) [reply docs/cli/usage/mock.md:20]: this URL wasn't working for me, added the Java 16 page in its place

**Review by josefaidt — CHANGES_REQUESTED** [review]
Hey @danabrams thanks again for taking this on! I've added a few comments for adjustments

**Comment by danabrams** [comment]
> Hey @danabrams thanks again for taking this on! I've added a few comments for adjustments

Both suggestions are fair enough. I've committed them.

**Review by josefaidt — APPROVED** [review]

**Review by ammarkarachi — APPROVED** [review]

**Pull request merged**